embed/etcd.go

```diff
@@ -426,9 +426,13 @@ func (e *Etcd) Close() {
 func stopServers(ctx context.Context, ss *servers) {
 	shutdownNow := func() {
 		// first, close the http.Server
-		ss.http.Shutdown(ctx)
+		if ss.http != nil {
+			ss.http.Shutdown(ctx)
+		}
 		// then close grpc.Server; cancels all active RPCs
-		ss.grpc.Stop()
+		if ss.grpc != nil {
+			ss.grpc.Stop()
+		}
```

Comment:
The test should always fail, why sometimes the workflow checks are all green?

Reply:
Let me check this.

Reply:
```diff
diff --git a/pkg/expect/expect.go b/pkg/expect/expect.go
index 2e32eb526..24dc575d4 100644
--- a/pkg/expect/expect.go
+++ b/pkg/expect/expect.go
@@ -55,7 +55,7 @@ func NewExpectWithEnv(name string, args []string, env []string) (ep *ExpectProce
        cmd.Env = env
        ep = &ExpectProcess{
                cmd:        cmd,
-               StopSignal: syscall.SIGKILL,
+               StopSignal: syscall.SIGTERM,
        }
        ep.cond = sync.NewCond(&ep.mu)
        ep.cmd.Stderr = ep.cmd.Stdout
```

Since we use `SIGKILL` in https://github.com/etcd-io/etcd/blob/4e800bdc6613cce4c67203b344a8959950ea1e3b/pkg/expect/expect.go#L58, and we didn't check the [clus.Close](https://github.com/etcd-io/etcd/blob/4e800bdc6613cce4c67203b344a8959950ea1e3b/tests/e2e/cmux_test.go#L69)'s return. If we switch it to `SIGTERM`, the log will show us panic.

```
../../bin/etcd-7830: 2023-07-07 08:23:36.526793 N | pkg/osutil: received terminated signal, shutting down...
../../bin/etcd-7830: panic: runtime error: invalid memory address or nil pointer dereference
../../bin/etcd-7830: [signal SIGSEGV: segmentation violation code=0x1 addr=0x130 pc=0x9ccd45]
../../bin/etcd-7830:
../../bin/etcd-7830: goroutine 180 [running]:
../../bin/etcd-7830: google.golang.org/grpc.(*Server).Stop(0x0)
../../bin/etcd-7830:    /home/fuwei/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1390 +0x45
../../bin/etcd-7830: go.etcd.io/etcd/embed.stopServers.func1()
../../bin/etcd-7830:    /home/fuwei/go/src/go.etcd.io/etcd/embed/etcd.go:431 +0x3c
../../bin/etcd-7830: go.etcd.io/etcd/embed.stopServers({0x115a558, 0xc000a52030}, 0xc000200738)
../../bin/etcd-7830:    /home/fuwei/go/src/go.etcd.io/etcd/embed/etcd.go:438 +0x7d
../../bin/etcd-7830: go.etcd.io/etcd/embed.(*Etcd).Close(0xc00014c600)
../../bin/etcd-7830:    /home/fuwei/go/src/go.etcd.io/etcd/embed/etcd.go:392 +0x835
../../bin/etcd-7830: go.etcd.io/etcd/pkg/osutil.HandleInterrupts.func1()
../../bin/etcd-7830:    /home/fuwei/go/src/go.etcd.io/etcd/pkg/osutil/interrupt_unix.go:70 +0x284
../../bin/etcd-7830: created by go.etcd.io/etcd/pkg/osutil.HandleInterrupts
../../bin/etcd-7830:    /home/fuwei/go/src/go.etcd.io/etcd/pkg/osutil/interrupt_unix.go:53 +0xce
=== RUN   TestConnectionMultiplexing/SeparateHTTP/ServerNonTLS
../../bin/etcd-7900: WARNING: Package "github.com/golang/protobuf/protoc-gen-go/generator" is deprecated.
../../bin/etcd-7900:    A future release of golang/protobuf will delete this package,
../../bin/etcd-7900:    which has long been excluded from the compatibility promise.
../../bin/etcd-7900:
../../bin/etcd-7900: 2023-07-07 08:23:36.536300 W | pkg/flags: unrecognized environment variable ETCD_UNSUPPORTED_ARCH=amd64
```